.github/workflows/testNginxForAzureDeployV1.yml

```diff
@@ -0,0 +1,69 @@
+# File: .github/workflows/testNginxForAzureDeployV1.yml
+
+name: Test Github action to update NGINX for Azure configurations and certificates
+on:
+  - push
+  - workflow_dispatch
+
+env:
+  NGINX_DEPLOYMENT_NAME: github-action-test-deployment
+  NGINX_TRANSFORMED_CONFIG_DIR_PATH: /etc/nginx/
+  NGINX_ROOT_CONFIG_FILE: nginx.conf
+  TEST_RESOURCE_GROUP_NAME: github-action-test-rg
+  NGINX_RESOURCE_LOCATION: westcentralus
+  NGINX_CERT_DETAILS: '[{"nginxCertName": "server4", "keyvaultSecret": "https://isolated-kv.vault.azure.net/secrets/server", "certificateVirtualPath": "/etc/ssl/certs/server4.crt", "keyVirtualPath": "/etc/ssl/certs/server4.key"  }, {"nginxCertName": "server5", "keyvaultSecret": "https://isolated-kv.vault.azure.net/secrets/acr-auth", "certificateVirtualPath": "/etc/ssl/certs/server5.crt", "keyVirtualPath": "/etc/ssl/certs/server5.key"  }]'
```

Comment:
> Can you explain why this input is is named "details" throughout instead of something like, `NGINX_CERTIFICATES`, or `NGINX_CERTS` if length is a concern?

No reason really. I had removed "details" in all my previous variable names and must have skipped this one. I have adjusted it though

> Can we merge the scripts together into a single `deploy.sh`? The majority of each script seems to be duplicate handling of options/arguments.

My aim in having different scripts is the separation of concerns and readability of code.
The duplicated bit is only the validation of the required variables (subscriptionID, resourceGroupName, deploymentName) which was meant to ensure the script is full proof regardless of the caller.